frontend/components/Main/MainHeader/AgentButton/index.tsx

```diff
@@ -0,0 +1,273 @@
+  );
+};
+
+const AgentNotRunningButton = () => {
+  const { wallets, masterSafeAddress } = useWallet();
+  const {
+    service,
+    serviceStatus,
+    setServiceStatus,
+    setIsServicePollingPaused,
+  } = useServices();
+  const { serviceTemplate } = useServiceTemplates();
+  const { showNotification } = useElectronApi();
+  const {
+    setIsPaused: setIsBalancePollingPaused,
+    safeBalance,
+    totalOlasStakedBalance,
+    totalEthBalance,
+  } = useBalance();
+  const { storeState } = useStore();
+  const { isEligibleForStaking, isAgentEvicted } = useStakingContractInfo();
+
+  const safeOlasBalance = safeBalance?.OLAS;
+  const safeOlasBalanceWithStaked =
+    safeOlasBalance === undefined || totalOlasStakedBalance === undefined
+      ? undefined
+      : safeOlasBalance + totalOlasStakedBalance;
+
+  const handleStart = useCallback(async () => {
+    // Must have a wallet to start the agent
+    if (!wallets?.[0]) return;
+
+    // Paused to stop overlapping service poll while wallet is created or service is built
+    setIsServicePollingPaused(true);
+
+    // Paused to stop confusing balance transitions while starting the agent
+    setIsBalancePollingPaused(true);
+
+    // Mock "DEPLOYING" status (service polling will update this once resumed)
+    setServiceStatus(DeploymentStatus.DEPLOYING);
+
+    // Create master safe if it doesn't exist
+    try {
+      if (!masterSafeAddress) {
+        await WalletService.createSafe(Chain.GNOSIS);
+      }
+    } catch (error) {
+      console.error(error);
+      setServiceStatus(undefined);
+      showNotification?.('Error while creating safe');
+      setIsServicePollingPaused(false);
+      setIsBalancePollingPaused(false);
+      return;
+    }
+
+    // Then create / deploy the service
+    try {
+      await ServicesService.createService({
+        serviceTemplate,
+        deploy: true,
+      });
+    } catch (error) {
+      console.error(error);
+      setServiceStatus(undefined);
+      showNotification?.('Error while deploying service');
+      setIsServicePollingPaused(false);
+      setIsBalancePollingPaused(false);
+      return;
+    }
+
+    // Show success notification based on whether there was a service prior to starting
+    try {
+      if (!service) {
+        showNotification?.('Your agent is now running!');
+      } else {
+        const minimumStakedAmountRequired =
+          getMinimumStakedAmountRequired(serviceTemplate);
+
+        showNotification?.(
+          `Your agent is running and you've staked ${minimumStakedAmountRequired} OLAS!`,
+        );
+      }
+    } catch (error) {
+      console.error(error);
+      showNotification?.('Error while showing "running" notification');
+    }
+
+    // Can assume successful deployment
+    // resume polling, optimistically update service status (poll will update, if needed)
+    setIsServicePollingPaused(false);
+    setIsBalancePollingPaused(false);
+    setServiceStatus(DeploymentStatus.DEPLOYED);
```

Comment:
can we rely on "(service polling will update this once resumed)" in this case?

Reply:
this is more an attempt to give immediate response after a successful deployment, by this point the service should be deploy successfully. though, i agree its not the best approach for synchronicity... would be better to call update function from useServices directly here, but any issue will be synced within five seconds of starting the service poll again

will update in further PR, good question